core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -265,40 +256,58 @@ func (eb *ethBroadcaster) handleInProgressEthTx(etx models.EthTx, attempt models
 	}
 
-	etx.BroadcastAt = &broadcastAt
+	etx.BroadcastAt = &initialBroadcastAt
 
-	if sendError.IsNonceTooLowError() || sendError.IsReplacementUnderpriced() {
-		if isVirginTransaction {
-			// If we get this error, and transaction has never been processed
-			// before, it is extremely likely that an external wallet messed with
-			// our nonce and we have no choice but to send it again.
-			//
-			// On the off chance an external wallet did not do this and we
-			// somehow got a network double-send or something, this fails by
-			// sending the transction twice (one will revert) which is a safe
-			// fail case.
-			return eb.handleExternalWalletUsedNonce(&etx, attempt)
-		}
-		// If this is resuming a previous crashed run, it is likely that our
-		// previous transaction was the one who was confirmed, in which case
-		// we hand it off to the eth confirmer to get the receipt.
+	if sendError.IsNonceTooLowError() {
+		// There are three scenarios that this can happen:
+		//
+		// SCENARIO 1
+		//
+		// This is resuming a previous crashed run. In this scenario, it is
+		// likely that our previous transaction was the one who was confirmed,
+		// in which case we hand it off to the eth confirmer to get the
+		// receipt.
+		//
+		// SCENARIO 2
 		//
-		// It is however possible an external wallet can have messed with the
-		// account during restart.
+		// It is also possible that an external wallet can have messed with the
+		// account and sent a transaction on this nonce.
 		//
 		// In this case, the onus is on the node operator since this is
 		// explicitly unsupported.
 		//
-		// If it turns out to have been an external wallet, this transaction
-		// will be retried on every new head (and will fail) until we reach out
-		// block depth target and then abandoned.
+		// If it turns out to have been an external wallet, we will never get a
+		// receipt for this transaction and it will eventually be marked as
+		// errored.
 		//
-		// Assume success and hand off to the ethConfirmer.
+		// The end result is that we will NOT SEND a transaction for this
+		// nonce.
+		//
+		// SCENARIO 3
+		//
+		// The network/eth client can be assumed to have at-least-once delivery
+		// behaviour. It is possible that the eth client could have already
+		// sent this exact same transaction even if this is our first time
+		// calling SendTransaction().
+		//
+		// In all scenarios, the correct thing to do is assume success for now
+		// and hand off to the eth confirmer to get the receipt (or mark as
+		// failed).
 		sendError = nil
 	}
 
+	if sendError.IsReplacementUnderpriced() {
+		// The only way we can have a replacement underpriced error is if we
+		// tried to submit a DIFFERENT transaction at this nonce than the one
+		// the remote node already has in its mempool (with an insufficiently
+		// increased gas price).
+		//
+		// The only way this can happen is if an external wallet used the account.
```

Comment:
Is this true? What if we tried to submit before and now the gas price is different?